i18n fallback rewrite generates empty pages with @astrojs/node adapter

### Astro Info

```block
Astro                    v6.1.1
Vite                     v7.3.1
Node                     v22.18.0
System                   macOS (arm64)
Package Manager          npm
Output                   static
Adapter                  @astrojs/node (v10.0.4)
Integrations             none
```

### If this issue only occurs in one browser, which browser is a problem?

N/A — build-time issue

### Describe the Bug

When using `i18n.fallback` with `fallbackType: 'rewrite'` and `@astrojs/node` adapter in middleware mode, all locale fallback pages are generated with empty response bodies during `astro build`. The default locale pages render correctly, but every fallback locale variant produces `(file not created, response body was empty)` and no files are written to disk.

**Config:**
```js
export default defineConfig({
    i18n: {
        locales: ['en', 'es', 'fr'],
        defaultLocale: 'en',
        fallback: { es: 'en', fr: 'en' },
        routing: {
            prefixDefaultLocale: false,
            fallbackType: 'rewrite',
        },
    },
    adapter: node({ mode: 'standalone' }), // also reproduces with 'middleware' mode
})
```

**Build output:**
```
 prerendering static routes
   ├─ /about/index.html (+4ms)
   ├─ /es/about/index.html (+1ms) (file not created, response body was empty)
   ├─ /fr/about/index.html (+0ms) (file not created, response body was empty)
   ├─ /index.html (+1ms)
   ├─ /es/index.html (+0ms) (file not created, response body was empty)
   ├─ /fr/index.html (+0ms) (file not created, response body was empty)
```

This affects all pages regardless of complexity — even a simple static page with just an `<h1>` produces empty fallback variants. Zero locale directories are created in the build output.

**Removing the adapter** (pure static output) makes fallback pages generate correctly. The bug only occurs when `@astrojs/node` adapter is present.

Pages have `export const prerender = true` set explicitly.

Note: This is related to #13964 / PR #14017 which fixed the runtime `ForbiddenRewrite` error for the same config combination. The runtime fix is present in v6.1.1, but the build-time generation of fallback pages produces empty bodies instead of the expected content.

### What's the expected result?

Locale fallback pages should be generated with the same content as the default locale page, matching Astro 5.x behavior. For example, `/es/about/index.html` should contain the same HTML as `/about/index.html`.

### Link to Minimal Reproducible Example

https://github.com/JefCurtis/astro-i18n-fallback-rewrite-repro

StackBlitz: https://stackblitz.com/github/JefCurtis/astro-i18n-fallback-rewrite-repro

Steps to reproduce:
1. Clone the repo
2. `npm install`
3. `npx astro build`
4. Observe that `/es/` and `/fr/` pages show `(file not created, response body was empty)`

### Participation

- [x] I am willing to submit a pull request for this issue.

I don't know if this is helpful but I ran into this issue and tried the patch suggested by @astrobot-houston and this caused another small issue with the dev server:


```
[ERROR] TypeError: this.result.serverIslandNameMap.get is not a function
      at ServerIslandComponent.getIslandContent (/Users/user/src/app/frontend/node_modules/astro/dist/runtime/server/render/server-islands.js:110:55)
      at ServerIslandComponent.init (/Users/user/src/app/frontend/node_modules/astro/dist/runtime/server/render/server-islands.js:52:32)
      at collectPropagatedHeadParts (/Users/user/src/app/frontend/node_modules/astro/dist/core/head-propagation/buffer.js:9:37)
      at async bufferPropagatedHead (/Users/user/src/app/frontend/node_modules/astro/dist/runtime/server/render/head-propagation/runtime.js:30:21)
      at async bufferHeadContent (/Users/user/src/app/frontend/node_modules/astro/dist/runtime/server/render/astro/render.js:126:3)
      at async renderToAsyncIterable (/Users/user/src/app/frontend/node_modules/astro/dist/runtime/server/render/astro/render.js:139:5)
      at async renderPage (/Users/user/src/app/frontend/node_modules/astro/dist/runtime/server/render/page.js:72:24)
      at async lastNext (/Users/user/src/app/frontend/node_modules/astro/dist/core/render-context.js:259:25)
      at async handleStaticRoute (file:///Users/user/src/app/frontend/node_modules/@sentry/astro/build/esm/server/middleware.js:64:30)
      at async eval (/Users/user/src/app/frontend/node_modules/astro/dist/i18n/middleware.js:26:22)
  17:05:35 [ERROR] this.result.serverIslandNameMap.get is not a function
```

I believe it's because I have a server island on my page. I'm not 100% sure this is the _correct_ fix but adding `this.serverIslands = await this.pipeline.getServerIslands();` https://github.com/withastro/astro/blob/main/packages/astro/src/core/render-context.ts#L500 seems to have done the trick.

--- a/.changeset/loose-groups-design.md
+++ b/.changeset/loose-groups-design.md
@@ -1,0 +1,5 @@
+---
+'astro': patch
+---
+
+Fixes an issue where `i18n.fallback` pages with `fallbackType: 'rewrite'` were emitted with empty bodies during `astro build`.

--- a/packages/astro/src/i18n/middleware.ts
+++ b/packages/astro/src/i18n/middleware.ts
@@ -102,7 +102,7 @@ export function createI18nMiddleware(
 		if (i18n.fallback && i18n.fallbackType) {
 			const fallbackDecision = computeFallbackRoute({
 				pathname: context.url.pathname,
-				responseStatus: response.status,
+				responseStatus: typeHeader === 'fallback' ? 404 : response.status,
 				currentLocale: context.currentLocale,
 				fallback: i18n.fallback,
 				fallbackType: i18n.fallbackType,

--- a/packages/astro/test/units/i18n/i18n-middleware.test.ts
+++ b/packages/astro/test/units/i18n/i18n-middleware.test.ts
@@ -22,6 +22,13 @@ function makePageResponse(
 	});
 }
 
+function makeFallbackSentinelResponse(): Response {
+	return new Response(null, {
+		status: 500,
+		headers: { 'X-Astro-Route-Type': 'fallback' },
+	});
+}
+
 interface I18nManifestOverrides {
 	defaultLocale?: string;
 	locales?: Locales;
@@ -187,6 +194,30 @@ describe('createI18nMiddleware', () => {
 
 			assert.equal(result.status, 200);
 			assert.equal(await result.text(), 'rewritten to /en/start');
+		});
+
+		it('rewrites for the fallback-type sentinel response (regression #16113)', async () => {
+			const handler = createI18nMiddleware(
+				makeI18nManifest({
+					strategy: 'pathname-prefix-other-locales',
+					fallbackType: 'rewrite',
+					fallback: { it: 'en' },
+				}),
+				'/',
+				'ignore',
+				'directory',
+			);
+			const ctx = createMockAPIContext({
+				url: 'http://localhost/it/about',
+				rewrite: async (_path: string) =>
+					new Response(`<h1>about page</h1>`, { status: 200 }),
+			} as any);
+			const next = async () => makeFallbackSentinelResponse();
+
+			const result = await callHandler(handler, ctx, next);
+
+			assert.equal(result.status, 200);
+			assert.match(await result.text(), /about page/);
 		});
 	});
 

